tests/baselines/reference/contextualTypeShouldBeLiteral.errors.txt

```diff
@@ -0,0 +1,130 @@
+tests/cases/compiler/contextualTypeShouldBeLiteral.ts(40,5): error TS2345: Argument of type '{ type2: "y"; value: "done"; method(): void; }' is not assignable to parameter of type 'X2 | Y2'.
+  Object literal may only specify known properties, but 'type2' does not exist in type 'X2'. Did you mean to write 'type1'?
+
+
+==== tests/cases/compiler/contextualTypeShouldBeLiteral.ts (1 errors) ====
+    interface X {
+        type: 'x';
+        value: string;
+        method(): void;
+    }
+    
+    interface Y {
+        type: 'y';
+        value: 'none' | 'done';
+        method(): void;
+    }
+    
+    function foo(bar: X | Y) { }
+    
+    foo({
+        type: 'y',
+        value: 'done',
+        method() {
+            this;
+            this.type;
+            this.value;
+        }
+    });
+    
+    interface X2 {
+        type1: 'x';
+        value: string;
+        method(): void;
+    }
+    
+    interface Y2 {
+        type2: 'y';
+        value: 'none' | 'done';
+        method(): void;
+    }
+    
+    function foo2(bar: X2 | Y2) { }
+    
+    foo2({
+        type2: 'y',
+        ~~~~~~~~~~
+!!! error TS2345: Argument of type '{ type2: "y"; value: "done"; method(): void; }' is not assignable to parameter of type 'X2 | Y2'.
+!!! error TS2345:   Object literal may only specify known properties, but 'type2' does not exist in type 'X2'. Did you mean to write 'type1'?
+        value: 'done',
+        method() {
+            this;
+            this.value;
+        }
+    });

```

Comment:
This is a new error - I'd argue this is actually a regression.

Reply:
Yeah, isn’t this perfectly assignable to `Y2`?